### PR TITLE
Imp/autoindex

### PR DESCRIPTION
**Pull request by pablopolyte**

### CODE REVIEW

File: src/Command/AddAutoIndexCommand.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                if (!copy($source, $newfile)) {
+                    echo "La copie $file du fichier a échoué...\n";
+                    $output->writeln('add file fail in ' . strtoupper($newfile));
+                }
+            }
+            $progress->advance();
+        }
+
+        $progress->finish();
+        $output->writeln('');
```
Comment (Poulinhoo) [inline src/Command/AddAutoIndexCommand.php:108]: Why leave it empty? If there is no need to delete this line or you can change with 
`
$io = new SymfonyStyle($input, $output);
`

see this link for know-how SymfonyStyle works: https://symfony.com/doc/current/console/style.html

File: README.md [hunk 2]
```diff
@@ -10,30 +10,12 @@ To use this script, choose one of the following options to get started:
...
-* Edit "index.php" file in "[sources](https://github.com/jmcollin/autoindex/tree/master/sources)" directory
+* Edit "index.php" file in "[sources](https://github.com/PrestaShopCorp/autoindex/assets/)" directory
 
 ## Usage
 
-- php-cli: `php index.php ../ps/modules/mymodules/ ../ps/themes/mythemes/`
-* Web browser:
-  - `http://localhost/autoindex/`
-  - `http://localhost/autoindex/?path=../ps/modules/mymodules/,../ps/themes/mythemes/`
+- php-cli: `php bin/autoindex prestashop:add:in (/your/module/path)`
```
Comment (zalexki) [inline README.md:17]: isn't the command called `prestashop:add:index` ?

File: bin/autoindex [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+foreach ($autoloadFiles as $autoloadFile) {
+    if (file_exists($autoloadFile)) {
+        require_once $autoloadFile;
+    }
+}
+
+use PrestaShop\AutoIndex\Command\AddAutoIndexCommand;
+use Symfony\Component\Console\Application;
+
+
```
Comment (zalexki) [inline bin/autoindex]: too much space here

File: bin/autoindex [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+$application = new Application('autoindex', '1.1.0');
+
```
Comment (zalexki) [inline bin/autoindex]: can remove this one too

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{
+    "name": "presta-shop-corp/autoindex",
+    "description": "Automatically add an 'indexphp' in all your directories or your zip file recursively",
```
Comment (zalexki) [inline composer.json]:
```suggestion
    "description": "Automatically add an 'index.php' in all the current or specified directories and all sub-directories.",
```

File: src/Command/AddAutoIndexCommand.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * 2007-2010 PrestaShop and Contributors
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php:3]: @Quetzacoalt91 should we run header stamp on this repo and apply AFL ?

File: src/Command/AddAutoIndexCommand.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        if ($dir === false) {
+            throw new \Exception('Could not get current directory. Check your permissions.');
+        }
+
+        $finder = new Finder();
+        $finder
+            ->directories()
+            ->in($dir)
+            ->exclude($this->filters);
+        $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: unused variable

File: src/Command/AddAutoIndexCommand.php [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+use PhpParser\ParserFactory;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: unused class

File: src/Command/AddAutoIndexCommand.php [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        $finder
+            ->directories()
+            ->in($dir)
+            ->exclude($this->filters);
+        $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+
+        $output->writeln('Updating directories in ' . strtoupper($dir) . ' folder ...');
+        $progress = new ProgressBar($output, count($finder));
+        $progress->start();
+        $progress->setRedrawFrequency(20);
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: do you really need/want to set this ?

**Review by zalexki — CHANGES_REQUESTED** [review]

File: src/Command/AddAutoIndexCommand.php [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+        foreach ($finder as $file) {
+            $newfile = $file->getRealPath() . '/index.php';
+            if (!file_exists($newfile)) {
+                if (!copy($source, $newfile)) {
+                    echo "La copie $file du fichier a échoué...\n";
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: why use `echo` when you use the `$output` just below ? 🤔

File: README.md [hunk 11]
```diff
@@ -14,7 +14,7 @@ Use your own "index.php" file
 
 ## Usage
 
-- php-cli: `php bin/autoindex prestashop:add:in (/your/module/path)`
+- php-cli: `php prestashop:add:in (/your/module/path)`
```
Comment (zalexki) [inline README.md]: are you sure ?

File: composer.json [hunk 12]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "presta-shop-corp/autoindex",
-    "description": "Automatically add an 'indexphp' in all your directories or your zip file recursively",
+    "description": "Automatically add an 'index.php' in all the current  or specified directories and all sub-directories.",
```
Comment (zalexki) [inline composer.json]: too much space

**Review by zalexki — CHANGES_REQUESTED** [review]

File: src/Command/AddAutoIndexCommand.php [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2010 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\AutoIndex\Command;
+
+use PhpParser\ParserFactory;
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: unused, please take time to properly review your own code before asking others this should be detected by your editor

File: src/Command/AddAutoIndexCommand.php [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+                    $output->writeln('add file fail in ' . strtoupper($newfile));
+                }
+            }
+            $progress->advance();
+        }
+
+        $progress->finish();
+        $output->writeln('');
+    }
+}
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: missing EOL

File: src/Command/AddAutoIndexCommand.php [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->filters = explode(',', $input->getOption('exclude'));
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $realPath = $input->getArgument('real_path');
+        if ($realPath) {
```
Comment (zalexki) [inline src/Command/AddAutoIndexCommand.php]: use strict condition

File: bin/autoindex [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+#!/usr/bin/env php
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
```
Comment (zalexki) [inline bin/autoindex]: outdated

File: bin/autoindex [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+use PrestaShop\AutoIndex\Command\AddAutoIndexCommand;
+use Symfony\Component\Console\Application;
+
+$application = new Application('autoindex', '1.1.0');
+$command = new AddAutoIndexCommand();
+
+$application->add($command);
+$application->setDefaultCommand($command->getName());
+$application->run();
```
Comment (zalexki) [inline bin/autoindex]: missing EOL

**Review by zalexki — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (pablopolyte) [reply README.md]: this : php bin/console prestashop:add:in

**Review by zalexki — APPROVED** [review]

**Review by Poulinhoo — APPROVED** [review]
For a next bet, it would be nice to use SymfonyStyle to replace the writeln :)

**Pull request merged**